.rustfmt.toml

```diff
@@ -1,67 +1,74 @@
-max_width = 90 # changed
-hard_tabs = false
-tab_spaces = 4
-newline_style = "Auto"
-use_small_heuristics = "Default"
-indent_style = "Block"
-wrap_comments = false
-format_code_in_doc_comments = false
-comment_width = 80
-normalize_comments = true # changed
-normalize_doc_attributes = false
-# license_template_path = "LICENSE_TEMPLATE" # changed
-format_strings = false
-format_macro_matchers = false
-format_macro_bodies = true
+# Most of the changed settings had defaults made for backcompat with earlier editions.
+# Brief descriptions are provided for nontrivial changes, otherwise a simple "changed" is
+# used.
+
+# `rustfmt` internal settings
+#
+# emit_mode = "Files"
+# make_backup = false
+
+binop_separator = "Front"
+blank_lines_lower_bound = 0
+blank_lines_upper_bound = 1
+brace_style = "SameLineWhere"
+color = "Auto"
+combine_control_expr = false # better all around in our opinions
+comment_width = 80 # changed
+condense_wildcard_suffixes = true # changed
+control_brace_style = "AlwaysSameLine"
+disable_all_formatting = false
+edition = "2018" # changed
 empty_item_single_line = true
-struct_lit_single_line = true
+enum_discrim_align_threshold = 0
+error_on_line_overflow = true # be more strict
+error_on_unformatted = true # be more strict
```

Comment:
> Error if Rustfmt is unable to get all lines within max_width, except for comments and string literals. If this happens, then it is a bug in Rustfmt.

Reeeeeally not sure if I want our CI to fail because of a bug in rustfmt ...

Reply:
There was a bug I encountered (https://github.com/rust-lang/rustfmt/issues/3805), but it was easily fixable by changing the signature width.